src/fields/ObjectField.js

```diff
@@ -0,0 +1,287 @@
+      {props.description && (
+        <DescriptionField
+            id={`${props.idSchema.$id}__description`}
+            description={props.description}
+            formContext={props.formContext} />
+      )}
+      {props.properties.map(prop => prop.content)}
+      {canExpand(props.schema, props.uiSchema, props.formData) && (
+        <AddButton
+            className="object-property-expand"
+            onClick={props.onAddClick(props.schema)}
+            disabled={props.disabled || props.readonly} />
+      )}
+    </fieldset>
+  );
+}
+
+class ObjectField extends Component {
+  static defaultProps = {
+    uiSchema: {},
+    formData: {},
+    errorSchema: {},
+    idSchema: {},
+    required: false,
+    disabled: false,
+    readonly: false,
+  };
+
+  state = {
+    wasPropertyKeyModified: false,
+    additionalProperties: {},
+  };
+
+  isRequired(name) {
+    const schema = this.props.schema;
+    return (
+      Array.isArray(schema.required) && schema.required.indexOf(name) !== -1
+    );
+  }
+
+  onPropertyChange = (name, addedByAdditionalProperties = false) => {
+    return (value, errorSchema) => {
+      if (!value && addedByAdditionalProperties) {
+        // Don't set value = undefined for fields added by
+        // additionalProperties. Doing so removes them from the
+        // formData, which causes them to completely disappear
+        // (including the input field for the property name). Unlike
+        // fields which are 'mandated' by the schema, these fields can
+        // be set to undefined by clicking a 'delete field' button, so
+        // set empty values to the empty string.
+        value = '';
+      }
+      const newFormData = { ...this.props.formData, [name]: value };
+      this.props.onChange(
+        newFormData,
+        errorSchema &&
+          this.props.errorSchema && {
+            ...this.props.errorSchema,
+            [name]: errorSchema,
+          }
+      );
+    };
+  };
+
+  onDropPropertyClick = (key) => {
+    return event => {
+      event.preventDefault();
+      const { onChange, formData } = this.props;
+      const copiedFormData = { ...formData };
+      delete copiedFormData[key];
+      onChange(copiedFormData);
+    };
+  };
+
+  getAvailableKey = (preferredKey, formData) => {
+    var index = 0;
+    var newKey = preferredKey;
+    while (formData.hasOwnProperty(newKey)) {
+      newKey = `${preferredKey}-${++index}`;
+    }
+    return newKey;
+  };
+
+  onKeyChange = (oldValue) => {
+    return (value, errorSchema) => {
+      if (oldValue === value) {
+        return;
+      }
+
+      value = this.getAvailableKey(value, this.props.formData);
+      const newFormData = { ...this.props.formData };
+      const newKeys = { [oldValue]: value };
+      const keyValues = Object.keys(newFormData).map(key => {
+        const newKey = newKeys[key] || key;
+        return { [newKey]: newFormData[key] };
+      });
+      const renamedObj = Object.assign({}, ...keyValues);
```

Comment:
I like `{ ...keyValues }` better. I feel like if you're using `Object.assign`, it should be with a non-empty object.

Reply:
no change required.

Reply:
This is just a copied component from RJSF core because I just needed a one line change to add the `registry` prop. I will not be making any changes to it. I made a PR to the RJSF project which has been approved and merged, but I don't know when it's going to be released.